CompilerSource/ipc/Sprite.fbs

```diff
@@ -0,0 +1,39 @@
+include "Point.fbs";
+include "Image.fbs";
+
+enum SpriteShape : byte { PRECISE, RECTANGLE, DISK, DIAMOND }
+
+enum SpriteBoundingBoxType : byte { AUTOMATIC, FULL_IMAGE, MANUAL }
+
+struct SpriteBoundingBox {
+  type:SpriteBoundingBoxType;
+  left:int;
+  right:int;
+  top:int;
+  bottom:int;
+}
+
+table Polygon {
+  points:[Point];  
+}
+
+table Sprite {
+  name:string;
+  id:int;
+
+  transparent:bool;
+  shape:SpriteShape;
+  alpha_tolerance:int;
+  separate_mask:bool;
+  smooth_edges:bool;

```

Comment:
Do we want to drop this field for both Sprites and Backgrounds? I always complained it never did anything in GM, and surprise surprise, it doesn't do anything in ENIGMA either. I don't think there's any graphical algorithm you can apply to 2D sprites to make them smoother, I don't think this flag is the same thing as interpolation. That itself is another problem because in older OpenGL's, interpolation is per-texture, whereas Direct3D has always been per-sampler (of which there are 8). Modern OpenGL has an extension to do it both ways.